### PR TITLE
Adding hardware addresses, oneliner logs, write ACKs, loop refactor

### PR DESCRIPTION
**Pull request by bitfliq**
Example of new oneline logging:
```
[11:28:47][I][econet:820]:   Write   : short: 0x380:0x1c0 [CONTROL_CENTER:FURNACE] : 01.01.04.07.00.00.48.43.4F.4D.4D.41.4E.44.00.00.00.01.45.00.00.00 (22)
[11:28:47][I][econet:789]:   AckMatch: short: 0x1c0:0x380 [FURNACE:CONTROL_CENTER] : 00
```

Main purpose of this is to enable grep/pattern detection of changing bytes.

### CODE REVIEW

**Comment by bitfliq** [comment]
Reviewing this, I see that I messed up the rebase and it modified commits you made. Happy to re-create without that garbage.

File: components/econet/econet.h [hunk 1]
```diff
@@ -172,6 +177,11 @@ class Econet : public Component {
 	float cc_blower_cfm = 0;
 	float cc_blower_rpm = 0;
 	float cc_dhumsetp = 0;
+	float cc_outdoor_air_temp = 0;
+    bool cc_dehumidifier_enabled = false;
```
Comment (Daniel-dev22) [inline components/econet/econet.h]: This is in my dev branch already.
https://github.com/Daniel-dev22/esphome-econet/blob/dev/components/econet/econet.h#L107

Would need to merge dev from here  into your branch.

File: components/econet/econet.h [hunk 2]
```diff
@@ -172,6 +177,11 @@ class Econet : public Component {
...
+	float cc_dehumidifier_drain = 0.0;
```
Comment (Daniel-dev22) [inline components/econet/econet.h]: Does this need the `.0` vs just `0`?

File: components/econet/econet.cpp [hunk 3]
```diff
@@ -241,25 +271,55 @@ void Econet::handle_text(uint32_t src_adr, std::string obj_string, std::string t
...
 		{
 			// data[0] = 4
 			// data[1 - 10] = 0
 			uint8_t heatcmd = data[11]; // 0 to 100 [0, 70, 100]
 			uint8_t coolcmd = data[12]; // [0, 2] Maybe 1 for 1st Stage?
-			uint16_t fan_cfm = (((uint16_t)data[13]) * 256) + data[14];
-			
+			uint16_t fan_cfm = convert_vector_to_uint16(13,data);
+			uint16_t fan_rpm = convert_vector_to_uint16(17,data);
+      uint16_t max_cfm = convert_vector_to_uint16(41,data);
```
Comment (Daniel-dev22) [inline components/econet/econet.cpp]: For some of the lines here the indentation is off.

File: components/econet/econet.h (hunk 2 above)
Comment (bitfliq) [reply components/econet/econet.h]: No, thanks for catching that.

File: components/econet/econet.h (hunk 1 above)
Comment (bitfliq) [reply components/econet/econet.h]: done

File: components/econet/econet.cpp (hunk 3 above)
Comment (bitfliq) [reply components/econet/econet.cpp]: thanks, fixed.

File: components/econet/econet.h [hunk 4]
```diff
@@ -219,13 +226,18 @@ class Econet : public Component {
...
 	uint32_t COMPUTER =      				192	;	// 80 00 00 C0
 	uint32_t FURNACE = 						0x1c0;	// 80 00 01 C0
-	uint32_t UNKNOWN_HANDLER =  			241	;	// 80 00 00 F1
 	uint32_t WIFI_MODULE =    				832	;	// 80 00 03 40
 	uint32_t SMARTEC_TRANSLATOR = 			4160;	// 80 00 10 40
 	uint32_t INTERNAL = 					4736; 	// 80 00 10 40
 	uint32_t HEAT_PUMP_WATER_HEATER =       0x1280; // 80 00 12 80
 	uint32_t AIR_HANDLER = 					0x3c0;	// 80 00 03 C0
 	uint32_t CONTROL_CENTER = 				0x380;	// 80 00 03 80
+	uint32_t ZONE_THERMOSTAT_2 =            0x680;
```
Comment (Daniel-dev22) [inline components/econet/econet.h:235]: I noticed this in the logs.
`0x380:0x400 [CONTROL_CENTER:UNKNOWN] : `

For me `0x400` is the heat pump. But I'm not sure if there's other traditional communicating 1 or 2 stage non heatpump air condensers out there so maybe we should add `OUTDOOR_UNIT` to be consistent with econet naming convention (`ODU_DEFR`,etc)

Comment (bitfliq) [reply components/econet/econet.h:235]: Sounds good to me.

**Comment by Daniel-dev22** [comment]
All the water heater data requests are marked as not supported. They are supported though. Maybe needs a tweak to the logic I didn't check yet how it determines what's supported.

[logs_econet-heatpump-water-heater_logs.txt](https://github.com/Daniel-dev22/esphome-econet/files/12266738/logs_econet-heatpump-water-heater_logs.txt)

File: components/econet/econet.cpp [hunk 5]
```diff
@@ -132,10 +132,15 @@ void Econet::handle_float(uint32_t src_adr, std::string obj_string, float value)
 		{
 			// Not Supported Yet
 		}
+    else
+    {
+ 			ESP_LOGI("econet", "  %s : not supported yet", obj_string.c_str());
```
Comment (Daniel-dev22) [inline components/econet/econet.cpp:137]: The indentation here appears off.

**Comment by Daniel-dev22** [comment]
Going to close this as it's going to generate massive merge conflicts at this time due to the major changes in #20. More than welcome to bring back a lot of the new additions for zones etc with a clean branch.

**Pull request closed without merging**